src/AWS/CRT/Internal/FFI.php

```diff
@@ -38,7 +40,24 @@ function __destruct() {
      * occurs at the CRT wrapper.
      */
     function __call(string $name, $args) {
-        return call_user_func_array(array(self::$ffi, $name), $args);
+        $transformed_args = [];
+        // process args, convert strings into byte buffers (uint8_t* buffer, size_t length)
+        $buffers = [];
+        foreach ($args as $arg) {
+            if (is_string($arg)) {
+                $buf_len = strlen($arg);
+                $buf = self::$ffi->new("uint8_t[" . $buf_len . "]");
+                \FFI::memcpy($buf, $arg, $buf_len);
+
+                array_push($buffers, $buf);
```

Comment:
Okay this is really nitpicky, but we have a standard in the SDK to prefer `$buffers []= $buf` over `array_push()`

Reply:
I don't like the standard, I think it's ugly and would prefer to move the spacing around, but it's there so I follow it